doc/user/pmd.rst

```diff
@@ -0,0 +1,205 @@
+.. _pm:
+
+**
+PM
+**
+
+:abbr:`PM (Path Monitoring)` is a daemon that provides path monitoring
+functionality to track connectivity and performance measurements linked
+with a remote IP address.
+
+.. _starting-pm:
+
+Starting PM
+===========
+
+Default configuration file for *pmd* is :file:`pmd.conf`.  The typical
+location of :file:`pmd.conf` is |INSTALL_PREFIX_ETC|/pmd.conf.
+
+If the user is using integrated config, then :file:`pmd.conf` need not be
+present and the :file:`frr.conf` is read instead.
+
+.. program:: pmd
+
+:abbr:`PM` supports all the common FRR daemon start options which are
+documented elsewhere.
+
+.. _using-pm:
+
+Using PM
+========
+
+All pm commands are under the enable node and preceeded by the ``pm``
+keyword.
+
+.. _pmd-commands:
+
+PMd Commands
+============
+
+.. index:: pm
+.. clicmd:: pm
+
+   Opens the PM daemon configuration node.
+
+.. index:: session <A.B.C.D|X:X::X:X> [{interface IFNAME|local-address <A.B.C.D|X:X::X:X>|vrf NAME}]
+.. clicmd:: session <A.B.C.D|X:X::X:X> [{interface IFNAME|local-address <A.B.C.D|X:X::X:X>|vrf NAME}]
+
+   Creates and configures a new PM session to send packets to and receive packets from.
+
+   `local-address` provides a local address that we should bind our
+   peer listener to and the address we should use to send the packets.
+
+   `interface` selects which interface we should use.
+
+   `vrf` selects which domain we want to use.
+
+.. index:: no session <A.B.C.D|X:X::X:X>$session [{local-address <A.B.C.D|X:X::X:X>$local|interface IFNAME$ifname|vrf NAME$vrfname}]
+.. clicmd:: no session <A.B.C.D|X:X::X:X>$session [{local-address <A.B.C.D|X:X::X:X>$local|interface IFNAME$ifname|vrf NAME$vrfname}]
+
+    Stops and removes the selected session.
+
+.. index:: show pm [vrf NAME] sessions [json]
+.. clicmd:: show pm [vrf NAME] sessions [json]
+
+    Show all configured PM sessions information and current status.
+
+.. index:: show pm [vrf NAME$vrfname] session <A.B.C.D|X:X::X:X>$peer [json]
+.. clicmd:: show pm [vrf NAME$vrfname] session <A.B.C.D|X:X::X:X>$peer [json]
+
+    Show status for a specific PM session.
+
+
+.. _pm-session-config:
+
+Session Configurations
+----------------------
+
+.. index:: [no] packet-size (1-65535)
+.. clicmd:: [no] packet-size (1-65535)
+
+   Configures the packet size of packets to send within sessions.
+   Default value is 80 bytes.
+
+.. index:: [no] packet-tos (1-255)
+.. clicmd:: [no] packet-tos (1-255)
+
+   The type of service value to be written in the appropriate
+   DSCP field of IP packet. Default value is 0xc0 which stands
+   for Inter Network Control value.
+
+.. index:: [no] interval (1-65535)
+.. clicmd:: [no] interval (1-65535)
+
+   Interval between each emission. This value is the interval
+   in milliseconds between each emission of packet. Default
+   interval value is set to 5000 ms.
```

Comment:
this is as aggressive as BFD is today. If everything remains same and encap is the only difference here then we better merge it with BFD to avoid code duplication. ICMP and BFD are services this process can offer to routing protocols.